tests/testthat/test-carehomes-vaccination.R

```diff
@@ -1,7 +1,9 @@
 context("carehomes (vaccination)")
 
-test_that("there are no infections if everyone is vaccinated with a vaccine
-          preventing 100% of acquisition", {
+test_that("No infections with perfect vaccine", {
```

Comment:
There is no test in this file that the new parameter `rel_p_hosp_if_sympt` behaves in a reasonable way - can you please add one?

Reply:
I have not added one because `rel_p_sympt` and `rel_p_hosp_if_sympt` have the exact same format and constraints as `rel_susceptibility` so I have used a single series of test [here](https://github.com/mrc-ide/sircovid/blob/0f8d38495a4bcc302fe0d78e66b723f3ccc203af/tests/testthat/test-utils.R#L109), [here](https://github.com/mrc-ide/sircovid/blob/0f8d38495a4bcc302fe0d78e66b723f3ccc203af/tests/testthat/test-utils.R#L131) and [here](https://github.com/mrc-ide/sircovid/blob/0f8d38495a4bcc302fe0d78e66b723f3ccc203af/tests/testthat/test-utils.R#L146). 

These tests are relative to functions `check_rel_param ` and `build_rel_param` which are called [here](https://github.com/mrc-ide/sircovid/blob/0f8d38495a4bcc302fe0d78e66b723f3ccc203af/R/carehomes.R#L634), in the same way for each of the three parameters. 

Is that an ok approach?

I will add another test for `build_rel_param` to ensure it does what we expect.

Reply:
In meant in the model, not the parameter validation - does it have a reasonable effect on the flows

Reply:
done [here](https://github.com/mrc-ide/sircovid/blob/268a7136050c0467c51ff7570bc9dbf98cbbbae5/tests/testthat/test-carehomes-vaccination.R#L40) and [here](https://github.com/mrc-ide/sircovid/blob/268a7136050c0467c51ff7570bc9dbf98cbbbae5/tests/testthat/test-carehomes-vaccination.R#L75)

Reply:
@richfitz upon adding these tests I realised I needed to ensure that rel_susceptibility, rel_p_sympt and rel_p_hosp_if_sympt have the same dimension. I have added some code to do this and some tests to check that when this is not the case, an error is thrown. Could you perhaps look at the PR again to make sure you are happy with these changes? 
By doing so I seem to have introduced some failing checks that I don't understand though.

Reply:
all solved now